sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/decimalExpressions.scala

```diff
@@ -168,23 +176,31 @@ case class CheckOverflowInSum(
   override def eval(input: InternalRow): Any = {
     val value = child.eval(input)
     if (value == null) {
-      if (nullOnOverflow) null else throw QueryExecutionErrors.overflowInSumOfDecimalError
+      if (nullOnOverflow) null
+      else throw QueryExecutionErrors.overflowInSumOfDecimalError(origin.context)
     } else {
       value.asInstanceOf[Decimal].toPrecision(
         dataType.precision,
         dataType.scale,
         Decimal.ROUND_HALF_UP,
-        nullOnOverflow)
+        nullOnOverflow,
+        origin.context)
     }
   }
 
   override protected def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
     val childGen = child.genCode(ctx)
+    val errorContextCode = if (nullOnOverflow) {
+      ctx.addReferenceObj("errCtx", origin.context)
+    } else {
+      "null"
```

Comment:
shall we pass empty string as that's the default parameter value?

Reply:
sure, updated